commit 150784f9285e656373cf3953ef4a7663f1e1a0f2
Author: Andrzej Hajda <andrzej.hajda@intel.com>
Date:   Tue Mar 14 16:19:20 2023 +0100

    drm/i915/gt: perform uc late init after probe error injection
    
    Probe pseudo errors should be injected only in places where real errors
    can be encountered, otherwise unwinding code can be broken.
    Placing intel_uc_init_late before i915_inject_probe_error violated
    this rule, resulting in following bug:
    __intel_gt_disable:655 GEM_BUG_ON(intel_gt_pm_is_awake(gt))
    
    Fixes: 481d458caede ("drm/i915/guc: Add golden context to GuC ADS")
    Acked-by: Nirmoy Das <nirmoy.das@intel.com>
    Reviewed-by: Andi Shyti <andi.shyti@linux.intel.com>
    Signed-off-by: Andrzej Hajda <andrzej.hajda@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20230314151920.1065847-1-andrzej.hajda@intel.com
    (cherry picked from commit c4252a11131c7f27a158294241466e2a4e7ff94e)
    Signed-off-by: Jani Nikula <jani.nikula@intel.com>

diff --git a/drivers/gpu/drm/i915/gt/intel_gt.c b/drivers/gpu/drm/i915/gt/intel_gt.c
index f0dbfc434e07..40d357cf8b04 100644
--- a/drivers/gpu/drm/i915/gt/intel_gt.c
+++ b/drivers/gpu/drm/i915/gt/intel_gt.c
@@ -737,12 +737,12 @@ int intel_gt_init(struct intel_gt *gt)
 	if (err)
 		goto err_gt;
 
-	intel_uc_init_late(&gt->uc);
-
 	err = i915_inject_probe_error(gt->i915, -EIO);
 	if (err)
 		goto err_gt;
 
+	intel_uc_init_late(&gt->uc);
+
 	intel_migrate_init(&gt->migrate, gt);
 
 	goto out_fw;